### ISSUE TITLE
After 8 hours of super slow processing it announces maybe my processor won't work. Why not at beginning?

### ISSUE THREAD

**Issue by tack32** [Anjok07/ultimatevocalremovergui#411]
I tried to split a 1 hour wav file and just extract the drum. It took 8 hours and then said it had a problem. Is there a solution
![failed memory message](https://user-images.githubusercontent.com/93066797/217435195-4aae035b-0620-479b-9275-ae7f94f891bb.jpg)
Last Error Received:

Process: Demucs

The application was unable to allocate enough system memory to use this model.

Please do the following:

1. Restart this application.
2. Ensure any CPU intensive applications are closed.
3. Then try again.

Please Note: Intel Pentium and Intel Celeron processors do not work well with this application.

If the error persists, the system may not have enough RAM, or your CPU might not be supported.

Raw Error Details:

RuntimeError: "[enforce fail at C:\actions-runner\_work\pytorch\pytorch\builder\windows\pytorch\c10\core\impl\alloc_cpu.cpp:72] data. DefaultCPUAllocator: not enough memory: you tried to allocate 155058176 bytes."
Traceback Error: "
  File "UVR.py", line 4592, in process_start
  File "separate.py", line 418, in seperate

These are the laptops specs:
Processor	Intel(R) Core(TM) i7-4710HQ CPU @ 2.50GHz, 2501 Mhz, 4 Core(s), 8 Logical Processor(s)
BIOS Version/Date	American Megatrends Inc. G551JM.205, 2/13/2015
SMBIOS Version	2.7
Embedded Controller Version	255.255
BIOS Mode	UEFI
BaseBoard Manufacturer	ASUSTeK COMPUTER INC.
BaseBoard Model	Not Available
BaseBoard Name	Base Board
Platform Role	Mobile
Secure Boot State	On
PCR7 Configuration	Binding Not Possible
Windows Directory	C:\Windows
System Directory	C:\Windows\system32
Boot Device	\Device\HarddiskVolume1
Locale	United States
Hardware Abstraction Layer	Version = "10.0.15063.909"
User Name	Asus\Asus
Time Zone	Central Standard Time
Installed Physical Memory (RAM)	16.0 GB
Total Physical Memory	15.9 GB
Available Physical Memory	11.1 GB

**Comment by Derpiesaurus** [comment]
This is because you ran out of RAM. 16GB may be enough for most models, and most games, but 32GB is probably necessary for such a large file to process on that model. 

It's likely it failed after 8 hours because there was a sudden jump in the usage of RAM that ended up going over your system budget, this could be because it has all of the chunks of the file it's processed in memory, and when it added another chunk to process, it just tipped it over the 16GB mark. It's possible if you closed all applications and then tried, it could work, but I'm not sure if I'd bet 8 more hours on it.

TLDR: Closing all applications, trying a different model and upgrading the amount of RAM you have I believe are the only solutions unfortunately.

**Comment by tack32** [comment]
Appreciate your reply. I doubt there's much chance to upgrade the memory
for this Asus Rog G551JM
Could you suggest a UVR5 model that's not so resource hungry? Thanks

**Comment by Derpiesaurus** [comment]
Ah I see, I hadn't realised you were on a laptop!

I'm not very sure what model is most RAM efficient, but since you're trying to extract the drums, I think you're pretty constrained to the Demucs models. I've got even less experience with those, and I sort of doubt there's all that much of a difference between them anyway, when it comes to RAM.

I think your best option is to try to split the file you want to process in half, or in quarters, and process each of those individually, then compile them back together. As for how you'd go about doing that, I suggest FFMPEG, since it's the most lightweight, and most easily uninstall-able media editor I know of.

I'll go through the steps to cut a file up and recompile it using FFMPEG, as well as the installation, since it can be quite confusing if you've never used something like it before, here I'll show the steps (12 to be precise) to cut the audio file into two 30 minute parts:

### DOWNLOADING FFMPEG:

### 1:
Go to this website: https://ffmpeg.org/download.html
Then you'll want to click on / hover over the blue Windows logo, and then click on the "Windows builds from gyan.dev" hyperlink.
![1](https://user-images.githubusercontent.com/38544173/218192689-4b70d7db-3d09-4a1e-aeaf-83fd391374a6.png)

### 2:
Next, when you're on the "gyan.dev" website, you'll want to look for the "ffmpeg-git-essentials-7z" file, and then click on that to download it. (I have all the text highlighted as I've got a dark mode extension for Chrome and it occasionally messes up colours, making the text invisible).
![2](https://user-images.githubusercontent.com/38544173/218192911-f157a8ab-d516-4993-93a3-9996e987853c.png)

### 3:
When you've downloaded the zipped file, open it up and you should see a folder inside it called "ffmpeg-(date)-git-(version number?)-essentials_build", grab that folder and put it on your desktop, or somewhere you can easily access it temporarily.
![3](https://user-images.githubusercontent.com/38544173/218193023-3045167a-a43f-4ca2-b497-70ab882595ee.png)

### 4:
Once you've put the folder from the zipped file somewhere, open it up, and navigate into the "bin" sub-folder.
![4](https://user-images.githubusercontent.com/38544173/218193383-f03bff04-56de-4f15-9daf-44d42bc442eb.png)

### 5:
Now that you're in the "bin" sub-folder, you should see "ffmpeg.exe", "ffplay.exe" and "ffprobe.exe". The "ffmpeg.exe" file is what we're looking for, if the others aren't there then that's no problem.
![5](https://user-images.githubusercontent.com/38544173/218193511-bf0afa29-bd92-4c03-8161-74f5dcabd632.png)

### PRE-PROCESSING THE AUDIO FILES:

### 6:
Now you'll want to drag in the file you want to process with UVR (the 1 hour long one). Here I've named my one "example.mp3", but it doesn't matter what the name is, or if it is a .wav file (you can just replace the .mp3 with .wav, as well as the name).
![6](https://user-images.githubusercontent.com/38544173/218193678-480006ff-4b18-4cb1-87ec-acaf8bc4c1c0.png)

### 7:
To use FFMPEG you need a prompt window open, I prefer Windows PowerShell, as it's easier to open. While in the "bin" folder, SHIFT + Right Click in an empty space, and it should bring up a prompt menu with "Open PowerShell window here" in it. If you're on Windows 11, you may have to click "Show more options" to see it.
![7](https://user-images.githubusercontent.com/38544173/218194027-75e9cdf0-3c1c-4379-81ae-14c24f1c6245.png)

### 8:
Now you can use FFMPEG to cut the first 30 minutes of the audio file using this command:
`ffmpeg -i "example.mp3 [swap .mp3 with .wav if applicable]" -c:a copy -to 00:30:00 "first30Minutes.mp3 [swap .mp3 with .wav if applicable]"`. You should remove the comments I added in the **[]**.

![8](https://user-images.githubusercontent.com/38544173/218194303-6c44b087-918c-4f63-81d3-b0fc015ab709.png)

### 9:
Like the last step, we'll use a very similar command to split off the 30 minutes at the end of the audio file:
`ffmpeg -i "example.mp3 [swap .mp3 with .wav if applicable]" -c:a copy -ss 00:30:00 "second30Minutes.mp3 [swap .mp3 with .wav if applicable]"`. And much like the previous step you should remove the commented parts in the **[]**.

![9](https://user-images.githubusercontent.com/38544173/218194739-c8cd2216-47f9-4a03-9555-016f69d215f2.png)

### PROCESSING THE SPLIT AUDIO FILES:

### 10:
After those two commands, if you check back in the folder, you should see that two more files were created, in my case "first30Minutes.mp3", and "second30Minutes.mp3".
![10](https://user-images.githubusercontent.com/38544173/218195056-ce45c55d-9af2-42a4-9804-5ba267b68b74.png)
You'll want to take each of those two files and process them through UVR individually with the model you tried to use before. Since these are half the size, hopefully they'll take up half the amount of RAM, and take half the time.

### COMBINING THE TWO PROCESSED AUDIO FILES:

### 11:
Finally, once you've processed both of the files (you may want to rename them so you don't get them muddled up, but in this example I'll keep the same names), put both of them back in the "bin" folder with the "ffmpeg.exe" file in it, then, use this command in the PowerShell console:
`ffmpeg -i "concat:first30Minutes.mp3|second30Minutes.mp3" -i second30Minutes.mp3 -c:a copy combined_output.mp3`

![11](https://user-images.githubusercontent.com/38544173/218195433-a559a511-4225-494a-ae4f-216d1322a4d0.png)

### 12:
Now, you should have a full hour long audio file named "combined_output.mp3" (or ".wav" if you changed it), this should have combined both the files you processed.
![12](https://user-images.githubusercontent.com/38544173/218195564-75894c0e-f020-4925-ba81-7a4a5aa482f1.png)

And to finish off, once you're happy with the "combined_output" file, move that, and the original hour long file to a different folder, then you can delete the zipped file, and the ffmpeg folder you downloaded. So it'll be like you never downloaded it in the first place, then nothing will get cluttered up!



Let me know if you need help with anything else; and if this fails you can try and send the audio file to me through something like Google Drive and I can see if I can process it for you.

**Comment by tack32** [comment]
Hi
That's kind of you to outline those steps. Clearly you're way over my
technical level. Even though you laid them out well it would still be a
challenging process. So it's the length of this thing that makes it so
difficult? I didn't know that. I checked my laptop's ability to be upgraded
and see 16 gb memory is the best it can do. By the way, I barely have used
this but through my Reaper forum I did install a program called AVConvert.
https://www.audioworld.de/AudioWorld%20AVConvert_e.php  I think it does
what you're talking about in a way that a person not so familiar with some
of the steps can grasp more easily. Actually the file that I've been
fooling with is simply a little bit of entertainment that I wanted to try
for fun. I find announcers in tennis to be distracting as they often talk
over the points so I wanted to try and separate them. I've been trying a
number of the AI programs after learning about them a few weeks ago. RipX,
Stemroller, LALAL.AI and UVR5. I'm afraid I can't remember which one I used
on this file the only time it worked but it only did half of it. It worked
as I hoped it would after I realized that the AI interpreted the racquet
hitting the ball as drums. I cut the file into around 1/3 the size and put
the file in my google docs. I don't use Google Docs much..  If you were
interested in breaking it down to stems (at least the drum part,which is
all I think I need) how would I share it with you?  Thanks!

**Comment by Derpiesaurus** [comment]
Yep, the longer the file is, the more data it has to store in RAM. 

I'm happy to have a go at extracting the drum stem from the file. I presume you mean Google Drive when you're referring to Google Docs? To share a file you can right-click on it, click on "Share", and I believe you should be able to click on copy link, and just send that? 

I remember I had some trouble with permissions last time I tried Google Drive a year or so ago, so in that case, https://www.mediafire.com/ hasn't let me down so far, as an alternative (although I haven't checked that in a while either). It might be worth sending both for redundancy, but one will likely work anyway if that's easier.

No worries about the method I suggested, I've never really made a guide before so it was fun either way!

**Comment by tack32** [comment]
Thanks. I'm going to do that with a different one. I already watched the
one before. This one is quite long but if it's possible, great. I joined
Mediafire for it. As I said all I really need would traditionally be the
drums (where the racquet hits the ball) but as long as the voice is out I'm
good with any other sounds that might be there. Thanks!
https://www.mediafire.com/file/qp2r5zoqohqnt6q/tenn.mp3/file

**Comment by Derpiesaurus** [comment]
Hey,

I managed to do about 7 hours of the total ~9h30m. I was running into some "Assertion" errors towards the end and I wasn't able to get past them.

I split it up into hour long bits and each bit took about 25GB of RAM in case you were wondering!

Here's a Google Drive link to the combined version. It's 8 hours long, which I don't know if it should be, since I only think I processed 7 hours, so there may be a duplicate near the end. Combining that many files with FFMPEG had me a little confused, so hopefully not, but possible: https://drive.google.com/file/d/12Ee6RtIJdEUSf2W5YC5D56aC9VrlCzVG/view?usp=sharing

I'm also going to include this mediafire link to a .ZIP file of all the 1 hour sections I processed, so you can go through those instead, to avoid any duplicates. It's also a smaller download than the combined one: https://www.mediafire.com/file/m0ew3otb3vaje65/1h-7h_Drum_Stem.zip/file

**Comment by tack32** [comment]
Great! Very nice of you to give it a try. I'm downloading it now and will
let you know how it goes. Thanks!

**Comment by tack32** [comment]
So far it's working well! I can only look at it from the beginning since I
don't want to see the score or who won before I look at the match. It will
take me quite a while (maybe some days) to get through it (especially as
The Super Bowl is just about to start) but the sound is in sync at the
beginning and not hearing the announcers so far. This will be fun.Good
job!  I'll keep you posted. Thanks!

**Comment by booToToM** [comment]
> This is because you ran out of RAM. 16GB may be enough for most models, and most games, but 32GB is probably necessary for such a large file to process on that model.
> 
> EDIT: Although it's possible you could split the file you're processing in half (two 30 minute files) and then process those individually.

Well, I have the same issue with a 44,1 khz 16 bits  audio that's under 3 minutes long. It just crashes miserably.
